assets/js/monaco_editor_completion_provider.js

```diff
@@ -0,0 +1,49 @@
+/*
+Listen for editor mount and register completion provider
+Supports MonacoEditorComponent
+*/
+const editorMountedHandler = (ev) => {
+  // build a list of completions for the editor
+  const dataCompletionsName = ev.detail.hook.el.getAttribute(
+    "data-completions-name",
+  );
+  const completionsEl = document.querySelector(
+    `[name="${dataCompletionsName}"]`,
+  );
+
+  let completions = [];
+
+  if (completionsEl != null) {
+    completions = JSON.parse(completionsEl.value);
+  }
+
+  const editor = ev.detail.editor.standalone_code_editor;
+
+  function createDependencyProposals(range) {
+    return completions.map(function (name) {
+      return {
+        label: name,
+        kind: monaco.languages.CompletionItemKind.Module,
+        insertText: name,
+        range: range,
+      };
+    });
+  }
+
+  monaco.languages.registerCompletionItemProvider("sql", {
```

Comment:
hmm we aren't able to add in completions via a <script> or similar within the component template? not a fan of separating out the js needed for a lv component, adds some additional complexity as well when we need to hunt for the js related to the monaco editor.